Run npx puppeteer browsers install chrome
puppeteer browsers install [browser]

Download and install the specified browser. If successful, the command outputs the actual browser buildId that was installed and the absolute path to the browser executable (see --format).

Positionals:
  browser  Which browser to install <browser>[@<buildId|latest>]. `latest` will try to find the latest available build. `buildId` is a browser-specific identifier such as a version or a revision.  [string]

Options:
  --version       Show version number  [boolean]
  --help          Show help  [boolean]
  --platform      Platform that the binary needs to be compatible with.  [string] [choices: "linux", "linux_arm", "mac", "mac_arm", "win32", "win64"] [default: Auto-detected]
  --base-url      Base URL to download from  [string]
  --install-deps  Whether to attempt installing system dependencies (only supported on Linux, requires root privileges).  [boolean] [default: false]
  --format        Format to use for the output. Supported placeholders: {{browser}}, {{buildId}}, {{path}}, {{platform}}  [string] [default: "{{browser}}@{{buildId}} {{path}}"]

Examples:
  puppeteer install                             Install all pinned browsers
  puppeteer install chrome                      Install the pinned available build of the Chrome browser.
  puppeteer install chrome@latest               Install the latest available build for the Chrome browser.
  puppeteer install chrome@stable               Install the latest available build for the Chrome browser from the stable channel.
  puppeteer install chrome@beta                 Install the latest available build for the Chrome browser from the beta channel.
  puppeteer install chrome@dev                  Install the latest available build for the Chrome browser from the dev channel.
  puppeteer install chrome@canary               Install the latest available build for the Chrome Canary browser.
  puppeteer install chrome@115                  Install the latest available build for Chrome 115.
  puppeteer install chromedriver@canary         Install the latest available build for ChromeDriver Canary.
  puppeteer install chromedriver@115            Install the latest available build for ChromeDriver 115.
  puppeteer install chromedriver@115.0.5790     Install the latest available patch (115.0.5790.X) build for ChromeDriver.
  puppeteer install chrome-headless-shell       Install the latest available chrome-headless-shell build.
  puppeteer install chrome-headless-shell@beta  Install the latest available chrome-headless-shell build corresponding to the Beta channel.
  puppeteer install chrome-headless-shell@118   Install the latest available chrome-headless-shell 118 build.
  puppeteer install chromium@1083080            Install the revision 1083080 of the Chromium browser.
  puppeteer install firefox                     Install the latest nightly available build of the Firefox browser.
  puppeteer install firefox@stable              Install the latest stable build of the Firefox browser.
  puppeteer install firefox@beta                Install the latest beta build of the Firefox browser.
  puppeteer install firefox@devedition          Install the latest devedition build of the Firefox browser.
  puppeteer install firefox@esr                 Install the latest ESR build of the Firefox browser.
  puppeteer install firefox@nightly             Install the latest nightly build of the Firefox browser.
  puppeteer install firefox@stable_111.0.1      Install a specific version of the Firefox browser.
  puppeteer install firefox --platform mac      Install the latest Mac (Intel) build of the Firefox browser.

Error: All providers failed for chrome 148.0.7778.97:
  - DefaultProvider: The browser folder (/home/runner/.cache/puppeteer/chrome/linux-148.0.7778.97) exists but the executable (/home/runner/.cache/puppeteer/chrome/linux-148.0.7778.97/chrome-linux64/chrome) is missing
    at installWithProviders (/home/runner/work/GRQ-validation/GRQ-validation/node_modules/@puppeteer/browsers/lib/cjs/install.js:119:11)
    at async install (/home/runner/work/GRQ-validation/GRQ-validation/node_modules/@puppeteer/browsers/lib/cjs/install.js:129:12)
    at async #install (/home/runner/work/GRQ-validation/GRQ-validation/node_modules/@puppeteer/browsers/lib/cjs/CLI.js:351:9)
    at async Object.handler (/home/runner/work/GRQ-validation/GRQ-validation/node_modules/@puppeteer/browsers/lib/cjs/CLI.js:228:17)
Error: Process completed with exit code 1.
0s
0s

--- a/.github/workflows/a11y.yml
+++ b/.github/workflows/a11y.yml
@@ -56,7 +56,18 @@ jobs:
             fi
             sleep 1
           done
+      # pa11y-ci pulls in Puppeteer, but Puppeteer v22+ no longer downloads
+      # Chrome during install — the browser must be fetched explicitly, or the
+      # run dies with "Could not find Chrome (ver. ...)". Install pa11y-ci
+      # locally first so its own Puppeteer drives the download; that pins the
+      # Chrome build to the exact version Puppeteer resolves at launch (a bare
+      # `npx puppeteer ...` could resolve a different Puppeteer, and so a
+      # mismatched Chrome build).
+      - name: Install pa11y-ci
+        run: npm install --no-save pa11y-ci
+      - name: Install the Chrome build Puppeteer expects
+        run: npx puppeteer browsers install chrome
       # pa11y-ci has no --standard CLI flag (that is a pa11y option); the
       # standard and target URLs are configured in pa11yci.json instead.
       - name: Run pa11y-ci (WCAG 2.1 AA) over the dashboard pages
-        run: npx --yes pa11y-ci --config pa11yci.json
+        run: npx pa11y-ci --config pa11yci.json

--- a/.github/workflows/a11y.yml
+++ b/.github/workflows/a11y.yml
@@ -39,7 +39,12 @@ jobs:
       # actions/setup-node@v4
       - uses: actions/setup-node@48b55a011bda9f5d6aeb4c2d9c7362e8dae4041e
         with:
-          node-version: "20"
+          # Track the current LTS rather than a pinned major: Node 20 reaches
+          # end-of-life on GitHub-hosted runners (2026-09-16), so pinning it
+          # runs the toolchain (http-server, pa11y-ci) on an unsupported
+          # interpreter. lts/* keeps the gate on a maintained runtime, matching
+          # markdown-lint.yml (Issue #171).
+          node-version: "lts/*"
       - name: Serve docs/ dashboard and wait for it to come up
         run: |
           set -euo pipefail

--- a/docs/archive/pr-summaries/pr-summary-171.md
+++ b/docs/archive/pr-summaries/pr-summary-171.md
@@ -1,0 +1,52 @@
+# PR Summary — Issue #171
+
+## Summary
+
+The accessibility workflow `.github/workflows/a11y.yml` pinned its
+`actions/setup-node` step to `node-version: "20"`, an end-of-life-soon major.
+Node 20 is scheduled for force-upgrade and removal on GitHub-hosted runners
+(removal 2026-09-16), so pinning it runs the Node-only toolchain
+(`http-server`, `pa11y-ci`) on an unsupported interpreter that no longer
+receives security fixes.
+
+The step drives Node-only tooling, so a full Deno migration is not practical.
+The pragmatic fix tracks a maintained LTS — `node-version: "lts/*"` — matching
+what `markdown-lint.yml` already does, keeping the a11y gate on a supported
+runtime through and beyond the September 2026 Node 20 removal.
+
+Closes #171.
+
+### Deno regression avoided
+
+This is a Deno repo. The changed step runs Node-only tooling (`http-server`,
+`pa11y-ci`) that has no Deno equivalent here, so the existing
+`actions/setup-node` step is retained and simply re-pointed at `lts/*` — no new
+Node tooling, dependencies, or config were introduced.
+
+## Evidence
+
+Backend/CI change — no web UI to screenshot. Verified via the Deno test suite.
+
+```mermaid
+flowchart LR
+    A["node-version: 20 (EOL 2026-09-16)"] -->|fix| B["node-version: lts/* (maintained)"]
+    B --> C[a11y gate stays on a supported runtime]
+```
+
+Test run after the fix:
+
+```
+a11y workflow tracks a maintained Node runtime, not an EOL major (Issue #171) ... ok
+ok | 10 passed | 0 failed
+```
+
+Full suite: `ok | 257 passed (55 steps) | 0 failed`.
+
+## Test Plan
+
+- Added `tests/a11y_workflow_test.ts::"a11y workflow tracks a maintained Node
+  runtime, not an EOL major (Issue #171)"` — parses the workflow, locates the
+  `setup-node` step, and asserts `node-version` is not an EOL/EOL-soon major
+  (`18`/`19`/`20`/`21`). This test fails against the old `"20"` pin and passes
+  after the change to `"lts/*"`.
+- All existing `a11y_workflow_test.ts` cases remain unchanged and pass.

